src/main/java/zd/zero/waifu/motivator/plugin/onboarding/UserOnboarding.kt

```diff
@@ -15,6 +15,7 @@ object UserOnboarding {
 
     fun attemptToPerformNewUpdateActions() {
         getNewVersion().ifPresent { newVersion ->
+            startOnboarding()
```

Comment:
This is the simplest way I can think of to catch both scenarios, where we start the onboarding state when we found a new version.

I tested this with the ff steps:
Test Case 1: With the restart
1. Build 2 plugins locally with different versions (to simulate an update)
2. Start a new idea instance (from [intellij-community](https://github.com/JetBrains/intellij-community) repo)
3. Drag the 1st plugin to the idea instance–no sayonara
4. Restart the idea instance
5. Drag the 2nd plugin with a different version to simulate update–no sayonara

Test Case 2: Without restart
*Same steps from TC1 skipping step `4`*

Reply:
:thinking: 
I can't replicate the test cases because

```
2020-11-01 08:35:01,175 [  50969]   INFO - lij.ide.plugins.DynamicPlugins - Plugin zd.zero.waifu-motivator-plugin is not unload-safe because class loader cannot be unloaded 
```
Which requires me to restart the IDE. So there is no project listener installed yet.

I would have thought that this code would be called

https://github.com/waifu-motivator/waifu-motivator-plugin/blob/ad13b4b06ac848a864d7f187da7a60299a1b4f5f/src/main/java/zd/zero/waifu/motivator/plugin/listeners/PluginInstallListener.kt#L23

before this code:

https://github.com/waifu-motivator/waifu-motivator-plugin/blob/ad13b4b06ac848a864d7f187da7a60299a1b4f5f/src/main/java/zd/zero/waifu/motivator/plugin/WaifuMotivatorProject.java#L49

:question: 

Which would have cause the project closed to be executed before the project opened code. (You have to close before you can open again) So the flag would not be set. 

If it was calling the project opened on install we would not have had #198 

So I don't see how this works (in theory)

Reply:
It seems that whenever we update the plugin it requires a restart. IJ doesn't properly unload the plugin on the update. But on the first install, it properly works. Do you think that we can just catch the first installation as for each plugin updates it always restarts, create another ticket for properly handling the plugin update for listeners?

Reply:
> Do you think that we can just catch the first installation as for each plugin updates it always restarts, create another ticket for properly handling the plugin update for listeners?

If you are suggesting that we use what this PR provides and open a new issue if it actually is still an issue (after publishing), then yes, that works for me.